group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorLoader.java

```diff
@@ -46,6 +46,35 @@ public short unknownType() {
         }
     }
 
+    /**
+     * Object that is returned as part of the future from load(). Holds the partition load time and the
+     * end time.
+     */
+    class LoadSummary {
+        private final long loadTimeMs;
```

Comment:
How about storing startTime and endTime? Then we could compute the delta when requested? Could you also add the number of records and bytes read?